### PR TITLE
Speed up series inversion with delayed reduction (as we did for exp).

### PR DESCRIPTION
**Pull request by wbhart**

### CODE REVIEW

**Comment by thofma** [comment]
Looking at the test here https://github.com/Nemocas/AbstractAlgebra.jl/pull/828/checks?check_run_id=2218728284, I think we missed a depracation for `map_coeffs`? Maybe with the keyword argument?

**Comment by wbhart** [comment]
It's way worse than that. I actually have entirely the wrong deprecation for map_coeffs. Probably got distracted mid line of code.

I have the right deprecation and will add it now.

However, do we want to export map_coefficients? It seems we currently don't export it.

**Comment by wbhart** [comment]
My mistake, I was just misspelling it. Anyhow, I've added the correct deprecation to this PR. Should be fixed now.

**Comment by wbhart** [comment]
And the CI has died. :-(

File: src/Deprecations.jl [hunk 1]
```diff
@@ -47,5 +47,7 @@ end
 
 # Deprecated in 0.16.*
 
-@deprecate map_coeffs(f) map_coefficients(f)
+@deprecate map_coeffs(f, g::AbstractAlgebra.PolyElem; parent=parent(g)) map_coefficients(f, g; parent=parent)
```
Comment (thofma) [inline src/Deprecations.jl:50]: This also has a `cached` keyword argument I think (Oscar tests are failing because it seems to be missing).

Comment (wbhart) [reply src/Deprecations.jl]: Yeah I went by the documentation, which was wrong.

**Comment by wbhart** [comment]
That map_coeffs call doesn't look correct to me. It has too many arguments. Where is this function defined?

**Comment by thofma** [comment]
You mean https://github.com/Nemocas/AbstractAlgebra.jl/blob/master/src/generic/Poly.jl#L2708?

**Comment by wbhart** [comment]
Nevermind, I miscounted braces.

I wonder why it is still failing. I've added the deprecation.

**Comment by wbhart** [comment]
Oh I don't know. Now it is complaining about a missing coercion. I don't see how this PR has anything to do with that.

**Comment by wbhart** [comment]
Let's hope it is right this time. Thanks.

**Comment by wbhart** [comment]
This time for sure! :-)

**Comment by wbhart** [comment]
Oh dear, Julia now prints Array{T, 2} as Matrix{T}? This breaks all our doctests. What fun.

**Comment by thofma** [comment]
Let's merge this here first.

We build the documentation with version "1", which changed to julia 1.6 because of the new release. We can either pin it to use 1.5 or adjust the doctests. But that is something for a different PR.

**Pull request merged**